src/mame/drivers/dec8.cpp

```diff
@@ -948,6 +841,73 @@ static ADDRESS_MAP_START( dec8_mcu_io_map, AS_IO, 8, dec8_state )
 	AM_RANGE(MCS51_PORT_P0,MCS51_PORT_P3) AM_READWRITE(dec8_mcu_from_main_r, dec8_mcu_to_main_w)
 ADDRESS_MAP_END
 
+/*
+	Super Real Darwin is similar but only appears to have a single port
+*/
+
+READ8_MEMBER(dec8_state::srdarwin_mcu_from_main_r)
+{
+	uint8_t ret = 0;
+
+
+	switch (offset)
+	{
+		case 0:
+			ret = m_i8751_port0;
+			break;
+
+		case 1:
+			ret = 0x00;
+			logerror("%s: srdarwin_mcu_from_main_r %02x %02x\n", machine().describe_context(), offset, ret);
+			break;
+
+		case 2:
+			ret = 0xff;
+			break;
+		case 3:
+			ret = ioport("I8751")->read();
```

Comment:
Can you please use a `required_ioport` or `optional_ioport` state member for this?  Runtime tagmap lookups really hurt performance.